File: JavaScript/findOutlier.js
//You are given an array (which will have a length of at least 3, but could be very large) containing integers. 
//The array is either entirely comprised of odd integers or entirely comprised of even integers except for a single integer N. 
//Write a method that takes the array as an argument and returns this "outlier" N.

//solution

function findOutlier(integers) {
    var a = [],
        b = [];

    integers.map(function (val) {
        val % 2 === 0 ? a.push(val) : b.push(val);
    });

    return a.length === 1 ? a[0] : b[0];
}

//other solutions:
function findOutlier(integers){
    const even = integers.filter(int => int % 2 === 0);
    const odd  = integers.filter(int => int % 2 !== 0);
    return even.length === 1 ? even[0] : odd[0];
  }

  function findOutlier(integers){
    var res;
    var odd = integers.filter(function(e){
      if(e%2 != 0){
        res = e;
      };
      return e%2 == 0;
    });
    
    return odd.length>1 ? res : odd.pop();
  }

  function findOutlier(integers){
    return integers.slice(0,3).filter(even).length >=2 ? integers.find(odd) : integers.find(even);
  }
  function even(num){
    return (num % 2 == 0);
  }
  function odd(num){
    return !even(num)
  }

File: Java/findOutlier.java
// You are given an array (which will have a length of at least 3, 
//but could be very large) containing integers. 
//The array is either entirely comprised of odd integers or entirely 
//comprised of even integers except for a single integer N. 
//Write a method that takes the array as an argument and returns this "outlier" N.

// Examples
// [2, 4, 0, 100, 4, 11, 2602, 36]
// Should return: 11 (the only odd number)

// [160, 3, 1719, 19, 11, 13, -21]
// Should return: 160 (the only even number)

import java.util.Arrays;

public class FindOutlier {
    
	static int find(int[] integers) {
		int out = 0;

		int[] even = Arrays.stream(integers).filter(x -> (x & 1) == 0).toArray();
		if (even.length==1) out = even[0]; 

		int[] odd = Arrays.stream(integers).filter(x -> (x & 1) != 0).toArray();
		if (odd.length==1) out = odd[0];

		return out;
	}
}